### ISSUE TITLE
JAX-RS 2.0 @BeanParam annotation

### ISSUE THREAD

**Issue by gmtwc** [stoicflame/enunciate#134]
Does Enunciate 2.0 support JAX-RS 2.0 @BeanParam annotation?
I tried running Enunciate on the module which uses it and it produced no documentation for the @BeanParam related code.

Thanks

**Comment by stoicflame** [comment]
Thanks for the report. No support yet, but we can probably get it in.

**Comment by gmtwc** [comment]
Thanks, that would be great!

**Comment by stoicflame** [comment]
Fixed at 97099a6.

**Comment by gmtwc** [comment]
I've compiled the M4 locally and modified the JAX-RS example to use @BeanParam.
After I build the example I do not see any documentation produces for the  @BeanParam use case.
Am I missing something?

Thanks!


package com.webcohesion.enunciate.examples.jaxrsjackson.genealogy.data.api;

import com.webcohesion.enunciate.examples.jaxrsjackson.genealogy.data.Person;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.BeanParam;

/**
 * @author Ryan Heaton
 */
@Path("/persons")
public class PersonService {
   /**
    *
    * @param id  Person's id
    * @return
    */
  @GET
  @Path("{id}")
  public Person getPerson(@PathParam("id") String id) {
    return new Person();
  }

  @GET
  @Path("/")
  public Person getPersonQuery(@BeanParam PersonQuery personQuery) {
    return new Person();
  }
}


package com.webcohesion.enunciate.examples.jaxrsjackson.genealogy.data.api;

import javax.ws.rs.QueryParam;

public class PersonQuery
{
   /**
    * Query by name
    */
   @QueryParam("name")
   private String name;

   public String getName()
   {
      return name;
   }

   public void setName(String name)
   {
      this.name = name;
   }
}

**Comment by stoicflame** [comment]
Huh. I guess I should have tried this out myself before closing it ;-).

**Comment by stoicflame** [comment]
Hey @gmtwc, are you sure you don't see any new documentation? At 523b647, I added a new method to the "edge case" example and it showed up just fine.

![134-illustration](https://cloud.githubusercontent.com/assets/145838/9797890/d423ce90-57ba-11e5-805c-fc27906a8270.png)

**Comment by gmtwc** [comment]
It seem to be working for me now as well and I think the problem was caused by the build environment.
No matter what changes to the BeanParam source code I did the changes were not propagating to the Enunciate docs.  I guess this was caused by the build system not doing a complete cleanup and Java compiler finding the older artifacts on the classpath.  So I have to do a complete cleanup for this feature to work OK.

Thanks!
